src/net_processing.cpp

```diff
@@ -1993,7 +1994,7 @@ void PeerManagerImpl::BlockConnected(
 
     // The following task can be skipped since we don't maintain a mempool for
     // the ibd/background chainstate.
-    if (role == ChainstateRole::BACKGROUND) {
+    if (!m_initial_sync_finished || role == ChainstateRole::BACKGROUND) {
```

Comment:
this was picked in https://github.com/bitcoin/bitcoin/pull/34054, so i guess this can be closed again?